src/Raven.Client/ServerWide/DatabaseTopology.cs

```diff
@@ -153,6 +153,40 @@ public class DatabaseTopology
         public LeaderStamp Stamp;
         public bool DynamicNodesDistribution;
         public int ReplicationFactor = 1;
+        public List<string> PriorityOrder;
+
+        private readonly Random _random = new Random();
+        public int Shuffle(string _, string __)
+        {
+            return _random.Next(-100, 100);
+        }
+
+        public void ReorderMembers()
+        {
+            if (PriorityOrder == null || PriorityOrder.Count == 0)
+            {
+                Members.Sort(Shuffle);
+                return;
+            }
+
+            var members = new List<string>();
+            for (int i = 0; i < PriorityOrder.Count; i++)
+            {
+                var priorityNode = PriorityOrder[i];
+                if (Members.Contains(priorityNode))
+                    members.Add(priorityNode);
+            }
+
+            for (int i = 0; i < Members.Count; i++)
```

Comment:
Should we shuffle the other stuff, that isn't in the priority order?

Reply:
okay, so I shuffle the members any way